### ISSUE TITLE
finish bsd bindings

### ISSUE THREAD

**Issue by misson20000** [reswitched/libtransistor#6]
There are a number of functions we don't have bindings for yet. Here is a list of all the functions in `nn::socket::sf::IClient` and what we have support for.

- [ ] StartMonitoring (what is this? this seems like something Nintendo. perhaps related to GetResourceStatistics?)
- [x] Socket
- [ ] SocketExempt (again, something Nintendo?)
- [x] Open
- [x] Select
- [ ] Poll
- [ ] Sysctl
- [x] Recv
- [ ] RecvFrom
- [x] Send
- [x] SendTo
- [x] Accept
- [x] Bind
- [x] Connect
- [ ] GetPeerName
- [ ] GetSockName
- [ ] GetSockOpt
- [x] Listen
- [ ] Ioctl
- [ ] Fcntl
- [ ] SetSockOpt
- [ ] Shutdown
- [ ] ShutdownAllSockets
- [ ] Write
- [ ] Read
- [x] Close
- [ ] DuplicateSocket
- [ ] GetResourceStatistics
- [ ] RecvMMsg
- [ ] SendMMsg

Most of these aren't important to me, but I think we should implement `Select` at least.

Here's the definition straight from SwIPC:
```
Unknown5(u32, bytes<0x18>, buffer<unknown, 0x21, 0>, buffer<unknown, 0x21, 0>, buffer<unknown, 0x21, 0>) -> (u32, u32, buffer<unknown, 0x22, 0>, buffer<unknown, 0x22, 0>, buffer<unknown, 0x22, 0>)
```

Relevant signatures and definitions:
```
int select(int nfds, fd_set *restrict readfds,
    fd_set *restrict writefds, fd_set *restrict errorfds,
    struct timeval *restrict timeout);
```
```
struct timeval {
    long    tv_sec;         /* seconds */
    long    tv_usec;        /* microseconds */
};
```
See bionic for fd_set definitions: https://android.googlesource.com/platform/bionic/+/7943df62f70f686b0c77532f6617b47255d75763/libc/include/sys/select.h.

This is my best guess at what all the parameters are:
```
Select(u32 nfds, bytes<0x18> timeout, buffer<fd_set, 0x21, 0> readfds_in, buffer<fd_set, 0x21, 0> writefds_in, buffer<fd_set, 0x21, 0> errorfds_in) -> (u32 ret, u32 errno, buffer<fd_set, 0x22, 0> readfds_out, buffer<fd_set, 0x22, 0> writefds_out, buffer<fd_set, 0x22, 0> errorfds_out)
```
I'm not exactly sure how `struct timeval` can become 0x18 bytes long though.

For examples of PegaSwitch test cases, see ipcsock3.js: https://gist.github.com/misson20000/82f92209800787fedda758c4333c6c8c/

**Comment by hthh** [comment]
Your best guess looks good - I think the `struct timeval` is just padded with zero after the first two 64-bit values.

**Comment by misson20000** [comment]
implemented Select with 32b9391fe

**Comment by roblabla** [comment]
I think @kgsws managed to get Poll to work ?
